src/pods/generate/TemplatedResourcesGenerator.ts

```diff
@@ -112,11 +137,18 @@ export class TemplatedResourcesGenerator implements ResourcesGenerator {
   Promise<Record<string, { link: TemplateResourceLink; meta?: TemplateResourceLink }>> {
     const files = await fsPromises.readdir(folderPath);
     const links: Record<string, { link: TemplateResourceLink; meta?: TemplateResourceLink }> = { };
-    for (const name of files) {
+    for await (const name of files) {
       const link = await this.toTemplateLink(joinFilePath(folderPath, name), mapper);
       const { path } = link.identifier;
       links[path] = Object.assign(links[path] || {}, link.isMetadata ? { meta: link } : { link });
     }
+
+    // Temporary code to trace race condition https://github.com/solid/community-server/issues/1059
+    const expected = readdirSync(folderPath);
+    assert.equal(Object.keys(links).length, expected.length, `Bug #1059 @groupLinks\nExpected: ${
+      expected.join()}\nActual: ${
+      JSON.stringify({ folderPath, links }, null, '  ')}`);
```

Comment:
TIL that you can also put a string as third parameter instead of giving an integer of how many spaces you want.

Reply:
TIL you can put an integer.